crates/anvil/src/eth/miner.rs

```diff
@@ -272,3 +281,47 @@ impl fmt::Debug for ReadyTransactionMiner {
             .finish_non_exhaustive()
     }
 }
+
+/// This struct represents mining mode in a simplified, serializable format
+#[derive(Debug, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub enum SerializableMiningMode {
+    None,
+    Auto(SerializableReadyTransactionMiner),
+    FixedBlockTime(SerializableFixedBlockTimeMiner),
+    Mixed(SerializableReadyTransactionMiner, SerializableFixedBlockTimeMiner),
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq)]
+pub struct SerializableFixedBlockTimeMiner {
+    /// The interval this fixed block time miner operates with, in seconds
+    pub period: u64,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq)]
+pub struct SerializableReadyTransactionMiner {
+    /// how many transactions to mine per block
+    max_transactions: usize,
+}
+
+impl From<&MiningMode> for SerializableMiningMode {
+    fn from(value: &MiningMode) -> Self {
```

Comment:
I'd rather make this a function of `MiningMode` itself

Reply:
Understood, I will modify the code.